src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```
@@ -157,6 +159,87 @@ public boolean isPinned()
         return getTextChannel().clearReactionsById(getId());
     }
 
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote, User user)
+    {
+        return getTextChannel().removeReactionById(getIdLong(), emote, user);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String unicode)
+    {
+        return channel.removeReactionById(getId(), unicode);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String unicode, User user)
+    {
+        return getTextChannel().removeReactionById(getId(), unicode, user);
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(Emote emote)
+    {
+        Checks.notNull(emote, "Emote");
+
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().isEmote() && r.getReactionEmote().getEmote().equals(emote))
+            .findFirst().orElse(null);
+        return new ReactionPaginationActionImpl(reaction);
```

What if `reaction` is null?

---

Should throw an error now.